src/structures/aligned_segment.cpp

```diff
@@ -0,0 +1,135 @@
+#include "structures/aligned_segment.hpp"
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+
+int32_t AlignedSegment::get_reference_start() const
+{
+    return pos;
+}
+
+int32_t AlignedSegment::get_reference_end() const
+{
+    int32_t current_pos = pos;
+    for (auto [element_length, element_operation] : cig)
+    {
+        switch(element_operation.to_char())
+        {
+            case 'M':
+                current_pos += element_length;
+                break;
+            case 'D':
+                current_pos += element_length;
+                break;
+            case 'N':
+                current_pos += element_length;
+                break;
+            case 'X':
+                current_pos += element_length;
+                break;
+            case '=':
+                current_pos += element_length;
+                break;
```

Comment:
When all cases should have the same action, you can fall through multiple cases by omitting the `break;`.
```suggestion
            case 'M':
            case 'D':
            case 'N':
            case 'X':
            case '=':
                current_pos += element_length;
                break;
```
If you get warnings, use the `[[fallthrough]]` attribute. https://en.cppreference.com/w/cpp/language/attributes/fallthrough

What is the `default:` case here?